clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -2311,6 +2311,38 @@ static mlir::Value createLLVMBitOp(mlir::Location loc,
   return result;
 }
 
+class CIRObjSizeOpLowering
+    : public mlir::OpConversionPattern<mlir::cir::ObjSizeOp> {
+public:
+  using OpConversionPattern<mlir::cir::ObjSizeOp>::OpConversionPattern;
+
+  mlir::LogicalResult
+  matchAndRewrite(mlir::cir::ObjSizeOp op, OpAdaptor adaptor,
+                  mlir::ConversionPatternRewriter &rewriter) const override {
+    auto resTy =
+        getTypeConverter()->convertType(op.getType()).cast<mlir::IntegerType>();
```

Comment:
If you are forcing a cast to `mlir::IntegerType` than why you are calling `convertType`? It's cleaner if the `convertType` can solve it all for you, this should just be: `auto resTy = getTypeConverter()->convertType(op.getType())`. You could also rename `resTy` to `llvmResTy`.